src/gui/vsInit.cpp

```diff
@@ -0,0 +1,164 @@
+  obtaining a copy of this software and associated documentation
+  files (the "Software"), to deal in the Software without
+  restriction, including without limitation the rights to use,
+  copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the
+  Software is furnished to do so, subject to the following
+  conditions:
+
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+*/
+//*****************************************************************
+
+/**
+ * \file vsInit.cpp
+ * \author
+ * \date February 2023
+ */
+
+#include "vsInit.h"
+
+#include <QCommandLineParser>
+#include <QDir>
+#include <QSettings>
+
+QString VsInit::parseDeeplink(QCoreApplication* app)
+{
+    // Parse command line for deep link
+    QCommandLineParser parser;
+    QCommandLineOption deeplinkOption(QStringList() << QStringLiteral("deeplink"));
+    deeplinkOption.setValueName(QStringLiteral("deeplink"));
+    parser.addOption(deeplinkOption);
+    parser.parse(app->arguments());
+    if (parser.isSet(deeplinkOption)) {
+        return parser.value(deeplinkOption);
+    } else {
+        return QLatin1String("");
+    }
+}
+
+void VsInit::checkForInstance(QCoreApplication* app, QString& deeplink,
+                              QSharedPointer<VirtualStudio> vs)
+{
+    m_vs = vs;
+    // Create socket
+    m_instanceCheckSocket.reset(new QLocalSocket(app));
+    // End process if instance exists
+    QObject::connect(
+        m_instanceCheckSocket.data(), &QLocalSocket::connected, app,
+        [=]() {
+            // pass deeplink to existing instance before quitting
+            if (!deeplink.isEmpty()) {
+                QByteArray baDeeplink = deeplink.toLocal8Bit();
+                qint64 writeBytes     = m_instanceCheckSocket->write(baDeeplink);
+                m_instanceCheckSocket->flush();
+                m_instanceCheckSocket->disconnectFromServer();  // remove next
+
+                if (writeBytes < 0) {
+                    qDebug() << "sending deeplink failed";
+                }
+            }
+            emit QCoreApplication::quit();
+        },
+        Qt::QueuedConnection);
+    // Create instanceServer to prevent new instances from being created
+    void (QLocalSocket::*errorFunc)(QLocalSocket::LocalSocketError);
+#if (QT_VERSION < QT_VERSION_CHECK(5, 15, 0))
+    errorFunc = &QLocalSocket::error;
+#else
```

Comment:
VS GUI [requires Qt 5.15](https://github.com/jacktrip/jacktrip/issues/852) anyway, AFAIR, so this could be possibly removed, I think.